File: Exercicios/IntroducaoAProgramacaoOrientadaAObjetos/Java/Polimorfismo/SobrecargaDeMétodos/Desenho.java
package Exercicios.IntroducaoAProgramacaoOrientadaAObjetos.Java.Polimorfismo.SobrecargaDeMétodos;

public class Desenho {
    // Método sem parâmetro
    public void mostrar() {
        System.out.println(" ");
        for (int i = 0; i < 10; i++) {
            if (i != 9) {
                System.out.print("*" + ", ");
            } else {
                System.out.println("*" + ";");
            }
        }        
        System.out.println(" ");
    }

    // Método com um parâmetro
    public void mostrar(String simb) {
        System.out.println(" ");
        for (int i = 0; i < 10; i++) {
            if (i != 9) {
                System.out.print(simb + ", ");
            } else {
                System.out.println(simb + ";");
            }            
        }        
        System.out.println(" ");
    }

    // Método com dois parâmetro
    public void mostrar(String simb, int n) {
        System.out.println(" ");
        for (int i = 0; i < n; i++) {
            if (i != n - 1) {
                System.out.print(simb + ", ");
            } else {
                System.out.println(simb + ".");
            }
        }
        System.out.println(" ");
    }
}

File: Exercicios/IntroducaoAProgramacaoOrientadaAObjetos/Python/Polimorfismo/Sobrecarga de Métodos/Desenho.py
class Desenho:

    # Método sem parâmetro
    def mostrar(self, simb='*', n=10):
        print(" ")
        for i in range(n):
            if i != n - 1:
                print(simb + ", ", end="")
            else:
                print("*" + ";")
        print(" ")

    # Método com um parâmetro
    def mostrar_com_simb(self, simb):
        print(" ")
        for i in range(10):
            if i != 9:
                print(simb + ", ", end="")
            else:
                print(simb + ";")
        print(" ")

    # Método com dois parâmetros
    def mostrar_com_simb_e_n(self, simb, n):
        print(" ")
        for i in range(n):
            if i != n - 1:
                print(simb + ", ", end="")
            else:
                print(simb + ".")
        print(" ")